server/app/views/applicant/ApplicantLayout.java

```diff
@@ -197,12 +197,26 @@ private NavTag renderNavBar(Http.Request request, Optional<String> userName, Mes
 
     String displayUserName = ApplicantUtils.getApplicantName(userName, messages);
     return nav()
-        .withClasses("bg-white", "border-b", "align-middle", "p-1", "grid", "grid-cols-3")
-        .with(branding())
-        .with(maybeRenderTiButton(profile, displayUserName))
+        .withClasses("bg-white", "border-b", "align-middle", "p-1", "flex", "flex-row", "flex-wrap")
         .with(
-            div(getLanguageForm(request, profile, messages), authDisplaySection(userName, messages))
-                .withClasses("justify-self-end", "flex", "flex-row"));
+            div()
+                .with(branding())
+                .withClasses(
+                    "items-center", "place-items-center", "flex-shrink-0", "grow", "md:grow-0"))
+        .with(maybeRenderTiButton(profile, displayUserName).withClasses("grow-0", "md:grow"))
+        .with(
+            div()
+                .with(
```

Comment:
**nitpick (style)**: _very_ up to preference but I think it looks better the fewer `with()` clauses we have.

In the extreme, this could look like 
```java
nav(
    div(
        branding())
        .withClasses("items-center", "place-items-center", "flex-shrink-0", "grow", "md:grow-0"),
    maybeRenderTiButton(profile, displayUserName).withClasses("grow-0", "md:grow"),
    div(
        getLanguageForm(request, profile, messages),
        authDisplaySection(userName, messages))
    .withClasses(
                    "flex",
                    "flex-row",
                    "grow",
                    "md:grow-0",
                    "shrink-0",
                    "md:shrink",
                    "place-content-center"))
.withClasses("bg-white", "border-b", "align-middle", "p-1", "flex", "flex-row", "flex-wrap");
```

Although that might be too far the other way, since the classes end up removed from the div they apply to.

Reply:
Sure, did some but not all of that suggestion.